test/e2e_node/runtime_conformance_test.go

```diff
@@ -28,111 +32,319 @@ import (
+						Volumes:       testVolumes,
+						NodeName:      *nodeName,
+						Namespace:     namespace,
+					}
+					err = terminateContainer.Create()
+					Expect(err).NotTo(HaveOccurred())
+					testCContainers = append(testCContainers, terminateContainer)
+
+					Eventually(func() api.PodPhase {
+						_, phase, _ := terminateContainer.GetStatus()
+						return phase
+					}, retryTimeout, pollInterval).ShouldNot(Equal(api.PodPending))
+
+					var status api.ContainerStatus
+					By("it should get the expected 'RestartCount'")
+					Eventually(func() int32 {
+						status, _, _ = terminateContainer.GetStatus()
+						return status.RestartCount
+					}, retryTimeout, pollInterval).Should(BeNumerically(testStatus.RestartCountOper, testStatus.RestartCount))
+
+					By("it should get the expected 'Ready' status")
+					Expect(status.Ready).To(Equal(testStatus.Ready))
+
+					By("it should get the expected 'State'")
+					Expect(GetContainerState(status.State) & testStatus.State).NotTo(Equal(0))
+
+					By("it should be possible to delete [Conformance]")
+					err = terminateContainer.Delete()
+					Expect(err).NotTo(HaveOccurred())
+					Eventually(func() bool {
+						isPresent, err := terminateContainer.Present()
+						return err == nil && !isPresent
+					}, retryTimeout, pollInterval).Should(BeTrue())
 				}
-				err := containerCase.Create()
-				Expect(err).NotTo(HaveOccurred())
 			})
 
-			It("it should report its phase as 'succeeded' and get a same container [Conformance]", func() {
-				var container ConformanceContainer
-				var err error
-				Eventually(func() api.PodPhase {
-					container, err = containerCase.Get()
-					return container.Phase
-				}, retryTimeout, pollInterval).Should(Equal(api.PodSucceeded))
-				Expect(err).NotTo(HaveOccurred())
-				Expect(container).Should(CContainerEqual(containerCase))
+			AfterEach(func() {
+				for _, cc := range testCContainers {
+					cc.Delete()
+				}
 			})
+		})
 
-			It("it should be possible to delete [Conformance]", func() {
-				err := containerCase.Delete()
-				Expect(err).NotTo(HaveOccurred())
-				Eventually(func() bool {
-					isPresent, err := containerCase.Present()
-					return err == nil && !isPresent
-				}, retryTimeout, pollInterval).Should(BeTrue())
+		Context("when start a container that keeps running", func() {
+			var testCContainers []ConformanceContainer
+
+			BeforeEach(func() {
+				testCContainers = []ConformanceContainer{}
+			})
+
+			It("it should run with the expected status [Conformance]", func() {
+				testContainer := api.Container{
+					Image:           "gcr.io/google_containers/busybox",
+					Command:         []string{"sh", "-c", "while true; do echo hello; sleep 1; done"},
+					ImagePullPolicy: api.PullIfNotPresent,
+				}
+				testStatuss := []testStatus{
+					{"loop-always", api.RestartPolicyAlways, api.PodRunning, ContainerStateWaiting | ContainerStateRunning | ContainerStateTerminated, "==", 0, true},
```

Comment:
I remembered why I always use 'with the expected status' even in the case which mainly test restart count. I think 'restart count' is also one of the statuses. It should be set correctly.
In first case (restart count>4), we expected our container to be restarted 3 times. In the later cases, we don't expect our container to be restarted.